gpm_api/tests/test_utils/test_geospatial.py

```diff
@@ -42,3 +40,378 @@
+    expected_extent = (e[0] - 0.2, e[1] + 0.2, e[2] - 0.2, e[3] + 0.2)
+    returned_extent = geospatial.get_country_extent(country)
+    assert returned_extent == expected_extent
+
+    # Test invalid country
+    country = "Invalid"
+    with pytest.raises(ValueError):
+        geospatial.get_country_extent(country)
+
+    # Test typo in country name
+    country = "Afganistan"
+    with pytest.raises(ValueError) as exception_info:
+        geospatial.get_country_extent(country)
+
+    assert "Afghanistan" in str(exception_info.value.args[0])
+
+    # Test invalid country type
+    country = 123
+    with pytest.raises(TypeError):
+        geospatial.get_country_extent(country)
+
+
+def test_get_continent_extent(
+    continent_extent_dictionary: ExtentDictionary,
+) -> None:
+    """Test get_continent_extent"""
+
+    # Test valid continent
+    continent = "Africa"
+    expected_extent = continent_extent_dictionary[continent]
+    returned_extent = geospatial.get_continent_extent(continent)
+    assert returned_extent == expected_extent
+
+    # Test invalid continent
+    continent = "Invalid"
+    with pytest.raises(ValueError):
+        geospatial.get_continent_extent(continent)
+
+    # Test typo in continent name
+    continent = "Arica"
+    with pytest.raises(ValueError) as exception_info:
+        geospatial.get_continent_extent(continent)
+
+    assert "Africa" in str(exception_info.value.args[0])
+
+    # Test invalid continent type
+    continent = 123
+    with pytest.raises(TypeError):
+        geospatial.get_continent_extent(continent)
+
+
+def test_get_extent() -> None:
+    """Test get_extent"""
+
+    ds = xr.Dataset(
+        {
+            "lon": [-10, 0, 20],
+            "lat": [-30, 0, 40],
+        }
+    )
+
+    # Test without padding
+    expected_extent = (-10, 20, -30, 40)
+    returned_extent = geospatial.get_extent(ds)
+    assert returned_extent == expected_extent
+
+    # Test with float padding
+    padding = 0.1
+    expected_extent = (-10.1, 20.1, -30.1, 40.1)
+    returned_extent = geospatial.get_extent(ds, padding=padding)
+    assert returned_extent == expected_extent
+
+    # Test with padding exceeding bounds
+    padding = 180
+    expected_extent = (-180, 180, -90, 90)
+    returned_extent = geospatial.get_extent(ds, padding=padding)
+    assert returned_extent == expected_extent
+
+    # Test with tuple padding
+    padding = (0.1, 0.2)
+    expected_extent = (-10.1, 20.1, -30.2, 40.2)
+    returned_extent = geospatial.get_extent(ds, padding=padding)
+    assert returned_extent == expected_extent
+
+    # Test with invalid padding
+    with pytest.raises(TypeError):
+        geospatial.get_extent(ds, padding="invalid")
+    with pytest.raises(ValueError):
+        geospatial.get_extent(ds, padding=(0.1,))
+    with pytest.raises(ValueError):
+        geospatial.get_extent(ds, padding=(0.1, 0.2, 0.3))
+
+    # Test with object crossing dateline (not implemented)
```

Comment:
Yeah this is always a dilemma. The utility is thought to adjust the extent of a cartopy plot. 
I would raise an error suggesting to use: 
- `ccrs.PlateCarree(central_longitude=180)` as cartopy projection 
- and we could potentially provide the `extent` for such cartopy projection (just subtract -180 to the longitudes of the input extent, and recompute the extent)

https://stackoverflow.com/questions/59584276/cartopy-set-extent-with-central-longitude-180

Reply:
Perhaps we can add a `central_longitude=0` argument to `get_extent`?

Reply:
Mmh maybe it's a solution.  
It means we substract the central longitude to the input longitudes?

In the plotting functions, we can try to retrieve the extent. If it raises the `NotImplementedError`, we try with the `get_extent(central_longitude=180)` and to use the the `ccrs.PlateCarree(central_longitude=180)` projection. 

Maybe it's worth to leave it as it is now, and to adapt the function when we define a plotting test case for a swath crossing the dateline and figure out the work-around to plot it nicely ...